### PR TITLE
xint, eint fixes, unittests; also reasonable-size; also make run from file

### PR DESCRIPTION
**Pull request by rfree-d**

### CODE REVIEW

File: src/xint.hpp [hunk 1]
```diff
@@ -306,29 +350,26 @@ std::ostream& operator<<(std::ostream& ostr, safer_int<T> obj) {
...
- * or when converting it to other type.
- * Supported operations can be done between this type, and any of: other variables of same type (recommend),
- * or build-in integral types, or build-in float types, or the "naked" boost::multiprecision based on cpp_int.
- * The operations that can be done, are: ++ -- (pre and post), + - * /   += -= *= /=   negation  > < >= <= ==
- * and also construction and assigment.
- * Not all combinations are yet ideally tested, best use xint with int, and on same signess type (xint with signed types,
- * uxint with unsigned).
-*/
-typedef safer_int<basic_xint> xint;
+typedef safer_int<basic_xint64>  xint64;    ///< safe integer -  64 bit,   signed (see class #safer_int and topic xint for details)
```
Comment (robertoleksy) [inline src/xint.hpp]: use type alias

File: src/test/utils_check.cpp [hunk 2]
```diff
@@ -277,5 +284,87 @@ TEST(utils_check, try_cached_as_check_what_message_test) {
 		if( msg.find(test_sentence) == string::npos )
 			FAIL();
 	}
+}
+
+TEST(utils_check, reasonable_size_string_test) {
+	{
```
Comment (rfree-d) [inline src/test/utils_check.cpp]: smaller tests blocks; I think no need to make separate function of them

File: src/test/xint.cpp [hunk 3]
```diff
@@ -6,16 +6,125 @@
...
+/// What if we would only use boost's big int (cpp int, checked) directly:
+using t_bigint64s = boost::multiprecision::number<
+	boost::multiprecision::cpp_int_backend<64, 64,
+		boost::multiprecision::signed_magnitude, boost::multiprecision::checked, void> >;
+using t_bigint64u = boost::multiprecision::number<
+	boost::multiprecision::cpp_int_backend<64, 64,
+		boost::multiprecision::signed_magnitude, boost::multiprecision::checked, void> >;
+
+long double get_double() { return std::numeric_limits<long double>::max(); }
+uint64_t get64u() { return std::numeric_limits<uint64_t>::max(); }
```
Comment (rfree-d) [inline src/test/xint.cpp]: 🖊 instead, moved to namespace

File: src/test/xint.cpp [hunk 4]
```diff
@@ -6,16 +6,125 @@
...
+
+	EXPECT_FALSE ( A1i >=  Zx  );
+	EXPECT_TRUE  ( A1i >=  A2x );
+	EXPECT_TRUE  ( Zi  >=  A1x );
+}
+
+
+// ===========================================================================================================
+
+#if 1 // turn off v1
```
Comment (rfree-d) [inline src/test/xint.cpp]: 🆗

File: src/test/xint.cpp [hunk 5]
```diff
@@ -398,14 +525,14 @@ TEST(xint, some_use) {
 }
 
 TEST(xint, range_u_incr) {
-	typedef uxint T;
+	typedef xintu T;
```
Comment (rfree-d) [inline src/test/xint.cpp]: 🆗

File: src/test/xint.cpp [hunk 6]
```diff
@@ -398,14 +525,14 @@ TEST(xint, some_use) {
...
 	T a("0xFFFFFFFFFFFFFFFE");
 	EXPECT_NO_THROW( { a++; } );	EXPECT_EQ(a , T("0xFFFFFFFFFFFFFFFF"));
 	EXPECT_THROW( { a++; } , std::runtime_error );	EXPECT_EQ(a , T("0xFFFFFFFFFFFFFFFF"));
 	EXPECT_THROW( { a++; } , std::runtime_error );	EXPECT_EQ(a , T("0xFFFFFFFFFFFFFFFF"));
 }
 TEST(xint, range_u_decr) {
-	typedef uxint T;
+	typedef xintu T;
```
Comment (rfree-d) [inline src/test/xint.cpp]: 🆗

File: src/test/xint.cpp [hunk 7]
```diff
@@ -532,6 +659,7 @@ TEST(xint, safe_create_xint_assign) {
 #undef maxni
 
 
+#endif // turn off v1
```
Comment (rfree-d) [inline src/test/xint.cpp]: 🆗

File: src/test/xint_tmp.cpp [hunk 8]
```diff
@@ -10,6 +10,8 @@
 #if USE_BOOST_MULTIPRECISION
 
 
+#if 1 // turn off v1
```
Comment (rfree-d) [inline src/test/xint_tmp.cpp]: 🆗

File: src/xint.hpp [hunk 9]
```diff
@@ -14,33 +14,53 @@
...
-typedef boost::multiprecision::number<
-	boost::multiprecision::cpp_int_backend<64, 64,
-		boost::multiprecision::signed_magnitude, boost::multiprecision::checked, void> >
-basic_xint;
-
-typedef boost::multiprecision::number<
-	boost::multiprecision::cpp_int_backend<64, 64,
-		boost::multiprecision::unsigned_magnitude, boost::multiprecision::checked, void> >
-basic_uxint;
+typedef boost::multiprecision::number< boost::multiprecision::cpp_int_backend<
```
Comment (rfree-d) [inline src/xint.hpp]: 🆗

File: src/xint.hpp (hunk 1 above)
Comment (rfree-d) [inline src/xint.hpp]: 🆗

File: src/test/utils_check.cpp (hunk 2 above)
Comment (mikurys) [inline src/test/utils_check.cpp]: Test object are big, so all objects are created in separated scopes to free memory after test when object is not used anymore

**Comment by rfree-d** [comment]
Hm one submodule was not properly pushed then? Maybe just try merging it and fix if needed.

**Pull request merged**